apps/web-staking/src/server/services/Pool.service.ts

```diff
@@ -152,6 +152,23 @@ export function mapPool(pool: IPool): PoolInfo {
 	 *
 	 * @dev DB Socials structure: [website: string, twitter: string, discord: string, telegram: string, instagram: string, tiktok: string, youtube: string]
 	 */
+
+	// Set initial values from DB
+	let ownerShare = pool.ownerShare || 0;
+	let keyBucketShare = pool.keyBucketShare || 0;
+	let stakedBucketShare = pool.stakedBucketShare || 0;
+
+	// Determine if there is a pending share update
+	const nowInSeconds = Math.floor(Date.now() / 1000);
+	const updateSharesTimestamp = pool.updateSharesTimestamp || 0;
+
+	// If there is a pending share update, use the pending share values
+	if(updateSharesTimestamp > nowInSeconds) {
```

Comment:
Is this the correct order.

I think it should be reversed, the pending shares will take effect when updateSharesTimestamp is in the past so if `now > updateSharesTimestamp`

There should be a similar snippet in the `web3.service.ts` `toPoolInfo()`

Reply:
Its going to be important to check if `updateSharesTimestamp != 0`, then we can know that there has to be pendingShares as they will always be set together in `StakingPool.sol`

Reply:
I misunderstood, thank you for the clarification. Can you confirm if the updateSharesTimestamp is stored in seconds or ms?

Reply:
I think this is correct now. Thank you for pointing me to the reference in the other file. It gave me all the context I needed.